### PR TITLE
Create a class encrypt, create menu

### PR DESCRIPTION
**Pull request by AlexaZhuk**

### CODE REVIEW

File: AccountManager/Dates.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace AccountManager
+{
+	internal class Dates
```
Comment (MrSantiaga) [inline AccountManager/Dates.cs]: Dates -> UserCredentials

File: AccountManager/Encryptor.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+			using CryptoStream cs = new CryptoStream(ms, encryptor, CryptoStreamMode.Write);
+			using (StreamWriter sw = new StreamWriter(cs))
+			{
+				sw.Write(password);
+			}
+			byte[] encryptedPassword = ms.ToArray();
+			return Convert.ToBase64String(encryptedPassword);
+
+		}
+		private byte[] GetLogin(string login)
```
Comment (MrSantiaga) [inline AccountManager/Encryptor.cs]: Add blank line

File: AccountManager/Encryptor.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		{
+			byte[] loginBytes = Encoding.UTF8.GetBytes(login);
+			using SHA256 sha256 = SHA256.Create();
+			return sha256.ComputeHash(loginBytes);
+		}
+		private byte[] GetVector(string vector)
```
Comment (MrSantiaga) [inline AccountManager/Encryptor.cs:61]: Add blank line

File: AccountManager/Encryptor.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+namespace AccountManager
+{
+	internal class Encryptor : IDisposable
+	{
+		private Aes _aes;
+		private byte[] _vector;
+
+		public Encryptor() : this("ho-ho-ho")
+		{
+
```
Comment (MrSantiaga) [inline AccountManager/Encryptor.cs:13]: extra line

Comment (MrSantiaga) [reply AccountManager/Encryptor.cs:13]: {}

File: AccountManager/Program.cs [hunk 5]
```diff
@@ -4,7 +4,72 @@ internal class Program
...
 		{
-			// TODO
+			Menu();
+		}
+
+		static public void Menu()
+		{
+			Console.Clear();
+			Console.WriteLine("Welcome to Account Manager");
+			Console.WriteLine("Menu: \n1. - Add new credentails \n2. - List all credentiles \n3. - Find and show credentiles \n4. - Delete credentiles \n0. - Exit");
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: Better use multiple `Console.WriteLine` to increase readability

File: AccountManager/Program.cs [hunk 6]
```diff
@@ -4,7 +4,72 @@ internal class Program
...
+
+			bool tryParse = int.TryParse(Console.ReadLine(), out int choose);
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: tryParse -> isParsed

File: AccountManager/Program.cs [hunk 7]
```diff
@@ -4,7 +4,72 @@ internal class Program
...
+				Console.WriteLine("Unknown value. Try again");
+				int.TryParse(Console.ReadLine(), out choose);
+			}
+
+			switch (choose)
+			{
+				case 1:
+					{
+						Dates user = new Dates();
+						Encryptor first = new Encryptor();
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: first -> encryptor

File: AccountManager/Program.cs [hunk 8]
```diff
@@ -4,7 +4,72 @@ internal class Program
...
+							Console.WriteLine("Success");
+							user.Login = login;
+							user.Password = first.Encrypt(password, login);
+							Console.WriteLine($"User login is  {user.Login}  user password is {user.Password}");
+						}
+						else
+						{
+							Console.WriteLine("Password mismatch");
+						}
+						Console.WriteLine("Do you want to return to menu?\n 1. - Yes\n 2. - No");
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: Multiple `Console.WriteLine`

File: AccountManager/Program.cs [hunk 9]
```diff
@@ -4,7 +4,72 @@ internal class Program
...
+						int answerRetutn = int.Parse(Console.ReadLine());
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: typo

File: AccountManager/Program.cs [hunk 10]
```diff
@@ -4,7 +4,72 @@ internal class Program
...
+						if (answerRetutn == 1)
+						{
+							Menu();
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: Recusive call. Replace with loop

File: AccountManager/Program.cs [hunk 11]
```diff
@@ -4,7 +4,72 @@ internal class Program
...
+				case 1:
+					{
+						Dates user = new Dates();
+						Encryptor first = new Encryptor();
+						Console.WriteLine("Add your dates to sign. \nAdd your login");
+						string login = Console.ReadLine();
+						Console.WriteLine("Add your password");
+						string password = Console.ReadLine();
+						Console.WriteLine("Repeat your password, please");
+						string passwordRepeat = Console.ReadLine();
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: passwordRepeat -> passwordConfirmation

File: AccountManager/Dates.cs [hunk 12]
```diff
@@ -0,0 +1,9 @@
...
+	{
+		public string Login { get; set; }
+		public string Password { get; set; }
+		public List<Alllogins> allLogins { get; set; }
```
Comment (MrSantiaga) [inline AccountManager/Dates.cs]: NIT: Looks like wrong prototyping

File: AccountManager/Alllogins.cs [hunk 13]
```diff
@@ -0,0 +1,8 @@
+﻿namespace AccountManager
+{
+	internal class Alllogins
+	{
+		List<string> userLogins { get; set; }
```
Comment (MrSantiaga) [inline AccountManager/Alllogins.cs]: From upper case

File: AccountManager/Dates.cs (hunk 12 above)
Comment (MrSantiaga) [inline AccountManager/Dates.cs]: From upper case

File: AccountManager/Encryptor.cs [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+using System.Text;
+
+namespace AccountManager
+{
+	internal class Encryptor : IDisposable
+	{
+		private Aes _aes;
+		private byte[] _vector;
+
+		public Encryptor() : this("ho-ho-ho")
```
Comment (MrSantiaga) [inline AccountManager/Encryptor.cs]: NIT: Let's have more secure vector

File: AccountManager/Program.cs (hunk 7 above)
Comment (MrSantiaga) [inline AccountManager/Program.cs]: Encryptor is IDisposable

File: AccountManager/Program.cs (hunk 9 above)
Comment (MrSantiaga) [inline AccountManager/Program.cs]: Always use TryParse

File: AccountManager/Program.cs [hunk 15]
```diff
@@ -4,7 +4,72 @@ internal class Program
...
+			{
+				case 1:
+					{
+						Dates user = new Dates();
+						Encryptor first = new Encryptor();
+						Console.WriteLine("Add your dates to sign. \nAdd your login");
+						string login = Console.ReadLine();
+						Console.WriteLine("Add your password");
+						string password = Console.ReadLine();
+						Console.WriteLine("Repeat your password, please");
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: Repeat your password -> Confirm your password

File: AccountManager/Program.cs [hunk 16]
```diff
@@ -4,7 +4,72 @@ internal class Program
...
+			{
+				Console.WriteLine("Unknown value. Try again");
+				int.TryParse(Console.ReadLine(), out choose);
+			}
+
+			switch (choose)
+			{
+				case 1:
+					{
+						Dates user = new Dates();
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: user -> credentials

File: AccountManager/Program.cs [hunk 17]
```diff
@@ -4,7 +4,72 @@ internal class Program
...
+		}
+
+		static public void Menu()
+		{
+			Console.Clear();
+			Console.WriteLine("Welcome to Account Manager");
+			Console.WriteLine("Menu: \n1. - Add new credentails \n2. - List all credentiles \n3. - Find and show credentiles \n4. - Delete credentiles \n0. - Exit");
+
+			bool tryParse = int.TryParse(Console.ReadLine(), out int choose);
+			while ((!tryParse) || (choose < 0 || choose > 5))
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: Try to rework to use switch statement inside this loop

**Review by MrSantiaga — CHANGES_REQUESTED** [review]

File: AccountManager/Dates.cs (hunk 1 above)
Comment (AlexaZhuk) [reply AccountManager/Dates.cs]: OK

File: AccountManager/Encryptor.cs [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+			_vector = GetVector(vector);
+			_aes.IV = _vector;
+		}
+
+		public void Dispose()
+		{
+			_aes.Dispose();
+		}
+
+		public string Encrypt(string password, string key)
```
Comment (MrSantiaga) [inline AccountManager/Encryptor.cs]: password -> plainText

File: AccountManager/Program.cs [hunk 19]
```diff
@@ -1,10 +1,244 @@
...
+		static DataBase dataBase;
+		static string path = "C:\\Users\\Alexa\\Desktop\\test\\Dates.json";
 		static void Main(string[] args)
 		{
-			// TODO
+			dataBase = InitData();
+			Menu();
 		}
+
+		static public void Menu()
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: public static

File: AccountManager/Program.cs [hunk 20]
```diff
@@ -1,10 +1,244 @@
...
+									Console.WriteLine("Try again");
+									login = Console.ReadLine();
+								}
+							}
+							Console.WriteLine("Add your password");
+							string password = Console.ReadLine();
+							Console.WriteLine("Add your key");
+							string key = Console.ReadLine();
+							Console.WriteLine("Confirm your key, please");
+							string keyCofirm = Console.ReadLine();
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: typo

File: AccountManager/Program.cs [hunk 21]
```diff
@@ -1,10 +1,244 @@
...
+							}
+							else
+							{
+								Console.ForegroundColor = ConsoleColor.Red;
+								Console.WriteLine("Password mismatch");
+								Console.ResetColor();
+							}
+
+							PrintReturnMainMenu();
+							bool tryParse2 = int.TryParse(Console.ReadLine(), out int answerRetutn);
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: typo: answerRetutn

Comment (MrSantiaga) [reply AccountManager/Program.cs]: code duplication
move to while

File: AccountManager/Program.cs [hunk 22]
```diff
@@ -1,10 +1,244 @@
...
+							{
+								Console.ForegroundColor = ConsoleColor.Red;
+								Console.WriteLine("Uncorrect dates");
+								Console.ResetColor();
+								PrintReturnMainMenu();
+								tryParse2 = int.TryParse(Console.ReadLine(), out answerRetutn);
+							}
+							if (answerRetutn == 1)
+							{
+								continue;
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: Looks like invalid if statement

File: AccountManager/Program.cs [hunk 23]
```diff
@@ -1,10 +1,244 @@
...
+								Console.WriteLine(user.Login);
+							}
+							Console.WriteLine("Press key to continue");
+							Console.ReadKey();
+							break;
+						}
+					case 3:
+						{
+							Console.WriteLine("Enter your login");
+							string userLogin = Console.ReadLine();
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: NIT: userLogin -> login

File: AccountManager/Program.cs [hunk 24]
```diff
@@ -1,10 +1,244 @@
...
+				}
+			}
+		}
+
+		private static void PrintMainMenu()
+		{
+			Console.Clear();
+			Console.WriteLine("Welcome to Account Manager");
+			Console.WriteLine("Menu:");
+			Console.WriteLine("1. - Add new credentails ");
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: typo: credentials

File: AccountManager/Program.cs [hunk 25]
```diff
@@ -1,10 +1,244 @@
...
+			Console.WriteLine("2. - List all credentiles");
+			Console.WriteLine("3. - Find and show credentiles");
+			Console.WriteLine("4. - Delete credentiles");
+			Console.WriteLine("0. - Exit");
+		}
+
+		private static void PrintReturnMainMenu()
+		{
+			Console.WriteLine("Do you want to return to menu?\n 1. - Yes\n 2. - No");
```
Comment (MrSantiaga) [inline AccountManager/Program.cs:128]: Multiple `Console.WriteLine`

File: AccountManager/Program.cs [hunk 26]
```diff
@@ -1,10 +1,244 @@
...
+		}
+
+
+		public static void AddUserDatesJson(string login, string encryptedPassword)
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: AddUser

File: AccountManager/Program.cs [hunk 27]
```diff
@@ -1,10 +1,244 @@
...
+			{
+				string json = File.ReadAllText(path);
+				return JsonConvert.DeserializeObject<DataBase>(json);
+			}
+			dataBase = new DataBase();
+			dataBase.AllUserData = new List<UserData>();
+			return dataBase;
+		}
+
+		public static void GetSearch(string userLogin)
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: SearchByLogin

File: AccountManager/Program.cs [hunk 28]
```diff
@@ -1,10 +1,244 @@
...
+				{
+					Console.ForegroundColor = ConsoleColor.Red;
+					Console.WriteLine("This login is exist");
+					Console.ResetColor();
+					Console.WriteLine($"your login is {user.Login} password is {user.Password}");
+					Console.WriteLine("Do you want to decrypt your password? \n1.Yes \n2.No");
+					int answer = int.Parse(Console.ReadLine());
+					if (answer == 1)
+					{
+						using Encryptor first = new Encryptor();
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: first-> encryptor

File: AccountManager/Program.cs [hunk 29]
```diff
@@ -1,10 +1,244 @@
...
+						Console.WriteLine("Enter your key");
+						string key = Console.ReadLine();
+						try
+						{
+							string toDecrypt = first.Decrypt(user.Password, key);
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: decryptedPassword

File: AccountManager/Program.cs [hunk 30]
```diff
@@ -1,10 +1,244 @@
...
+					}
+					return;
+				}
+			}
+			Console.ForegroundColor = ConsoleColor.Red;
+			Console.WriteLine("This login is not founded");
+			Console.ResetColor();
+		}
+
+		public static void GetDelete()
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: Rework to have `DeleteByLogin(string login)`

File: AccountManager/Program.cs [hunk 31]
```diff
@@ -1,10 +1,244 @@
...
+				{
+					Console.WriteLine("This login is exist");
+					//Console.WriteLine($"your login is {user.Login} password is {user.Password}");
+					Console.ForegroundColor = ConsoleColor.Red;
+					Console.WriteLine($"Are you sure want to delete {user.Login}? \n1.Yes \n2.No");
+					Console.ResetColor();
+					int answer = int.Parse(Console.ReadLine());
+					if (answer == 1)
+					{
+						using Encryptor first = new Encryptor();
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: encryptor

Encryptor may be global

File: AccountManager/Program.cs [hunk 32]
```diff
@@ -1,10 +1,244 @@
...
+						Console.WriteLine("Enter your key");
+						string key = Console.ReadLine();
+						try
+						{
+							string toDecrypt = first.Decrypt(user.Password, key);
+							dataBase.AllUserData.Remove(user);
+							string resultTry = JsonConvert.SerializeObject(dataBase);
```
Comment (MrSantiaga) [inline AccountManager/Program.cs:215]: resultJson or serializedText

File: AccountManager/Program.cs [hunk 33]
```diff
@@ -1,10 +1,244 @@
...
+							File.WriteAllText(path, resultTry);
+							Console.WriteLine($"This user is deleted");
+						}
+						catch (Exception)
+						{
+							Console.ForegroundColor = ConsoleColor.Red;
+							Console.WriteLine("Your code is wrong. Try again");
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: Your key

File: AccountManager/Program.cs [hunk 34]
```diff
@@ -1,10 +1,244 @@
...
+					return;
+				}
+			}
+			Console.ForegroundColor = ConsoleColor.Red;
+			Console.WriteLine("This login is not founded");
+			Console.ResetColor();
+		}
+
+
+		public static void AddUser(string login, string password, string key)
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: why do we have `AddUserDatesJson` and `AddUser` with almost same logic?

**Review by MrSantiaga — CHANGES_REQUESTED** [review]

File: AccountManager/Encryptor.cs [hunk 35]
```diff
@@ -0,0 +1,77 @@
...
+
+		private byte[] GetVector(string vector)
+		{
+			byte[] keyBytes = Encoding.UTF8.GetBytes(vector);
+			using MD5 mD5 = MD5.Create();
+			return mD5.ComputeHash(keyBytes);
+		}
+	}
+}
+
```
Comment (MrSantiaga) [inline AccountManager/Encryptor.cs]: **EXTRA LINES**

File: AccountManager/Program.cs [hunk 36]
```diff
@@ -1,10 +1,255 @@
...
+		static DataBase dataBase;
+		static string path = "C:\\Users\\Alexa\\Desktop\\test\\Dates.json";
+		static Encryptor encryptor = new Encryptor();
+
 		static void Main(string[] args)
 		{
-			// TODO
+			dataBase = InitData();
+			Menu();
+		}
```
Comment (MrSantiaga) [inline AccountManager/Program.cs:15]: `encryptor.Dispose()`

File: AccountManager/Program.cs [hunk 37]
```diff
@@ -1,10 +1,255 @@
...
+
+		public static void GetDelete(string login)
+		{
+			foreach (UserData user in dataBase.AllUserData)
+			{
+				if (user.Login == login)
+				{
+					Console.WriteLine("This login is exist");
+					//Console.WriteLine($"your login is {user.Login} password is {user.Password}");
```
Comment (MrSantiaga) [inline AccountManager/Program.cs:202]: Remove

File: AccountManager/Program.cs [hunk 38]
```diff
@@ -1,10 +1,255 @@
...
+		private static void PrintReturnMainMenu()
+		{
+			Console.WriteLine("Do you want to return to menu?\n 1. - Yes\n 2. - No");
+			Console.WriteLine("1. - Yes");
+			Console.WriteLine("2. - No");
+		}
+
+		public static void AddUser(string login, string password, string key)
+		{
+			UserData user1 = new UserData();
```
Comment (MrSantiaga) [inline AccountManager/Program.cs]: user

**Review by MrSantiaga — CHANGES_REQUESTED** [review]

**Review by MrSantiaga — APPROVED** [review]

**Pull request merged**